src/htcondorce/rrd.py

```diff
@@ -213,3 +213,11 @@ def graph(environ, host, plot, interval):
 
     return graph
 
+
+def update(fname, value):
+    try:
+        rrdtool.update(fname, value)
+    except NameError:
+        cmd = ['/usr/bin/rrdtool', 'update', fname, value]
+        proc = subprocess.Popen(cmd)
+        proc.communicate()  # wait until 'rrdtool update' completes
```

Comment:
Does `rrdtool.update` return anything?  Because this rrd.update does not.

Also not sure if/how you want to handle non-success exit status from the rrdtool commandline.

Reply:
Not sure but we weren't checking for the return code before so /shrug

Reply:
Ok, in that case if the rrdtool module is present they'll behave the same.  Seems that rrdtool.update raises exceptions on errors though.  If you want to do the same for the command line version, you could replace your Popen().communicate() with check_call() .

Or maybe you don't care, in which case who can blame you.

Reply:
Yeah, that's a good suggestion so I'll drop in some `# TODO` comments but I'm not too worried about it for the time being